### PR TITLE
gossip: add missing dependencies

### PR DESCRIPTION
**Pull request by msanft**
## Description of changes

Supersedes #339552, fixes building Gossip after https://github.com/NixOS/nixpkgs/pull/214906

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/go/gossip/package.nix [hunk 1]
```diff
@@ -64,10 +65,12 @@ rustPlatform.buildRustPackage rec {
     openssl
   ] ++ lib.optionals stdenv.isDarwin [
     darwin.apple_sdk.frameworks.AppKit
+    darwin.apple_sdk.frameworks.Cocoa
     darwin.apple_sdk.frameworks.CoreGraphics
     darwin.apple_sdk.frameworks.Foundation
   ] ++ lib.optionals stdenv.isLinux [
     wayland
+    wayland-scanner
```
Comment (Aleksanaa) [inline pkgs/by-name/go/gossip/package.nix:73]: in `buildInputs`? `wayland-scanner` is used in build process I suppose?

Comment (msanft) [reply pkgs/by-name/go/gossip/package.nix:73]: Correct, thanks for noticing!

File: pkgs/by-name/go/gossip/package.nix [hunk 2]
```diff
@@ -47,32 +49,43 @@ rustPlatform.buildRustPackage rec {
...
+      fontconfig
+      libGL
+      libxkbcommon
+      openssl
+    ]
+    ++ lib.optionals stdenv.isDarwin [
+      darwin.apple_sdk.frameworks.AppKit
+      darwin.apple_sdk.frameworks.Cocoa
+      darwin.apple_sdk.frameworks.CoreGraphics
+      darwin.apple_sdk.frameworks.Foundation
```
Comment (Aleksanaa) [inline pkgs/by-name/go/gossip/package.nix:80]:
```suggestion
      darwin.apple_sdk.frameworks.Foundation
      darwin.apple_sdk.frameworks.ForceFeedback
```

**Comment by Mastermindaxe** [comment]
Still fails on `aarch64-darwin`:
[build_log_pr_341182.log](https://github.com/user-attachments/files/16963888/build_log_pr_341182.log)
Currently in a workshop. Can build and test, but can't really focus too much on it 😅

**Comment by Mastermindaxe** [comment]
Still fails on `aarch64-darwin`:
[build_log_pr_341182.log](https://github.com/user-attachments/files/16964081/build_log_pr_341182.log)
This time in the fixup phase and not in the build phase. Missing command `patchelf`

**Comment by reckenrode** [comment]
`postFixup` should be conditioned on `!stdenv.isDarwin`. `patchelf` doesn’t work with Mach-O, and that particular fixup probably isn’t needed on Darwin anyway.

**Comment by gshpychka** [comment]
Result of `nixpkgs-review pr 341182` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gossip</li>
  </ul>
</details>

File: pkgs/by-name/go/gossip/package.nix [hunk 3]
```diff
@@ -84,10 +99,20 @@ rustPlatform.buildRustPackage rec {
     ln -s $out/logo/gossip.png $out/share/icons/hicolor/128x128/apps/gossip.png
   '';
 
-  postFixup = ''
-    patchelf $out/bin/gossip \
-      --add-rpath ${lib.makeLibraryPath [ libGL libxkbcommon wayland ]}
-  '';
+  postFixup =
+    if !stdenv.isDarwin then
```
Comment (Aleksanaa) [inline pkgs/by-name/go/gossip/package.nix:103]: lib.optionalString (!stdenv.isDarwin) ''

**Comment by gshpychka** [comment]
Result of `nixpkgs-review pr 341182` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gossip</li>
  </ul>
</details>

**Review by gshpychka — APPROVED** [review]

**Pull request merged**